### PR TITLE
Use pagination helper to list pipelines in cli

### PR DESCRIPTION
**Pull request by xoxys**
This PR fixes the `woodecker-cli pipeline ls` command. Before this change it was no possible to list more than 50 pipelines as this is the hardcoded limit of the API for pipelines per page.

### CODE REVIEW

**Comment by xoxys** [comment]
Yes I know but this was already the case before thos change. 

Yes there is now server side filtering but not for all three options and I would now look into the open autogen client PR instead of copying everything manually to woodpecker-go again.

So can we merge this PR as is?

**Comment by xoxys** [comment]
You are right. Then lets keep it open for now and I will look into to clientgen first.

**Comment by 6543** [comment]
> ... not for all three options ...

the woodpecker-go currently does not have support for the new filters yes ... I'm also looking forward to generate it

**Comment by xoxys** [comment]
Depends-on: https://github.com/woodpecker-ci/woodpecker/pull/4494

File: woodpecker-go/woodpecker/repo.go [hunk 1]
```diff
@@ -77,6 +82,18 @@ func (opt *PipelineListOptions) QueryEncode() string {
...
 		query.Add("after", opt.After.Format(time.RFC3339))
 	}
+	if opt.Branch != "" {
+		query.Add("branch", opt.Branch)
+	}
+	if len(opt.Events) > 0 {
+		query.Add("event", strings.Join(opt.Events, ","))
+	}
+	if opt.RefContains != "" {
+		query.Add("ref_contains", opt.RefContains)
```
Comment (6543) [inline woodpecker-go/woodpecker/repo.go:92]:
```suggestion
		query.Add("ref", opt.RefContains)
```

**Review by 6543 — APPROVED** [review]

**Review by pat-s — APPROVED** [review]

**Pull request merged**